src/core/tests/any.cpp

```diff
@@ -520,3 +504,15 @@ TEST_F(AnyTests, accessUsingWrongBaseReference) {
     ASSERT_NO_THROW(p.as<WrongDerived>());
     ASSERT_THROW(p.as<WrongBase>(), ov::Exception);
 }
+
+TEST_F(AnyTests, ToString) {
+    Any p = 42;
+    ASSERT_TRUE(p.is<int>());
+    ASSERT_EQ("42", p.as<std::string>());
+}
+
+TEST_F(AnyTests, FromString) {
+    Any p = "42";
+    ASSERT_TRUE(p.is<std::string>());
+    ASSERT_EQ(42, p.as<int>());
+}
```

Comment:
should we add tests for bool which prints as YES / NO?

Reply:
added